packages/components/src/navigation/item/use-navigation-tree-item.js

```diff
@@ -8,19 +8,29 @@ import { useEffect } from '@wordpress/element';
  */
 import { useNavigationContext } from '../context';
 import { useNavigationMenuContext } from '../menu/context';
+import { normalizedSearch } from '../utils';
 
 export const useNavigationTreeItem = ( props ) => {
 	const {
+		activeMenu,
 		navigationTree: { addItem, removeItem },
 	} = useNavigationContext();
-	const { menu } = useNavigationMenuContext();
+	const { menu, search } = useNavigationMenuContext();
 
 	const key = props.item;
 	useEffect( () => {
-		addItem( key, { ...props, menu } );
+		const isMenuActive = activeMenu === menu;
+		const isItemVisible =
+			! search || normalizedSearch( props.title, search );
```

Comment:
I think this is fine for now, but I think we'll need to think of how to search from external dataset whose items are not completely included in navigation menu? For example, we might have Pages navigation with thousands of items, and we might not want to render them all within the menu.

Reply:
There is an `onSearch` prop that can be used to handle this kinds of things, e.g. to perform an actual database search.
(At least that's the idea, but I haven't tested it in a real-life scenario)